### PR TITLE
Create geekaditi-rat-in-a-maze-CPP.cpp

### PR DESCRIPTION
**Pull request by geekaditi**
Solution to the Rat in a maze problem in CPP (Recursion)

### CODE REVIEW

File: geekaditi-rat-in-a-maze-CPP.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+
+int main(){
+    char maze[5][5]={
+        "0000",
+        "00XX",
+        "0000",
+        "XX00"
+    };
```
Comment (vishal-burman) [inline geekaditi-rat-in-a-maze-CPP.cpp:59]: Hi @geekaditi Could you change the input from a hard-coded one to a generic input. The above hard-coded input can be put as a comment to show a particular example?

**Review by vishal-burman — CHANGES_REQUESTED** [review]

**Comment by vishal-burman** [comment]
Hi @geekaditi you can change the filename from **geekaditi-rat-in-a-maze-CPP.cpp**  to **rat-in-a-maze.cpp**.

**Comment by geekaditi** [comment]
> Hi @geekaditi you can change the filename from **geekaditi-rat-in-a-maze-CPP.cpp** to **rat-in-a-maze.cpp**.

I made the required changes. Please check now.

**Comment by vishal-burman** [comment]
@geekaditi The code is returning a segmentation fault. Do a cross-check.

File: rat-in-a-maze-CPP.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    sol[i][j] = 0;
+    return false;
+}
+
+
+int main(){
+    char maze[100000][100000];
+    int sol[100000][100000]={0};
```
Comment (vishal-burman) [inline rat-in-a-maze-CPP.cpp:55]:
```suggestion
char maze[100][100];
int sol[100][100]={0};
int main(){
```
You are overflowing the stack with a large declaration. Consider a smaller input constraint. Also declare the inputs outside of the main function if you choose larger constraints.

**Review by vishal-burman — CHANGES_REQUESTED** [review]

**Review by SidJain1412 — APPROVED** [review]

**Comment by SidJain1412** [comment]
@all-contributors please add @geekaditi for code

**Comment by shreayan98c** [comment]
Please add this program to the Recursion directory.

**Comment by vishal-burman** [comment]
@SidJain1412 Actually I suggested those changes and other parts of the program had to be modified because of those changes. The program is still not running. @geekaditi might need to debug the program based on those changes.

**Comment by vishal-burman** [comment]
@SidJain1412 I had some free time. I have fixed the program and added some modifications in #71

**Comment by SidJain1412** [comment]
Thanks @vishal-burman @shreayan98c , my bad

**Comment by SidJain1412** [comment]
@all-contributors please add @shreayan98c for review

**Pull request merged**